### PR TITLE
soc: esp32: add mcpwm support

### PR DESCRIPTION
**Pull request by LucasTambor**
Add new pwm driver based on the mcpwm peripheral.
Add test case for esp32 platform in the pwm_loopback test.
Add myself as codeowner for the ESP32 LEDC and MCPWM drivers.
Update hal_espressif to latest revision.

### CODE REVIEW

**Review by sylvioalves — DISMISSED** [review]

File: dts/bindings/pwm/espressif,esp32-mcpwm.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    type: int
+    required: true
+    description: |
+      8 bit timer prescale for the global clock.
+      Period of PWM_clk = 6.25ns * (PWM_CLK_PRESCALE + 1).
+
+  "#pwm-cells":
+    const: 3
+
+  prescale_timer0:
```
Comment (gmarull) [inline dts/bindings/pwm/espressif,esp32-mcpwm.yaml:101]: `prescale-timer0` (`-`, not `_`)

File: drivers/pwm/Kconfig.esp32 [hunk 2]
```diff
@@ -8,3 +8,9 @@ config PWM_LED_ESP32
 	help
 	  This option enables the PWM LED driver for ESP32 family of
 	  processors. Say y if you wish to use PWM LED port on ESP32.
+
+config MCPWM_ESP32
```
Comment (gmarull) [inline drivers/pwm/Kconfig.esp32:12]: please enable based on dt compat status being enabled (see e.g. gd32), and make it dependent on ESP32 SoCs. I can now select both ESP32 drivers even if compiling for other platforms... this needs to be fixed.

File: tests/drivers/pwm/pwm_loopback/testcase.yaml [hunk 3]
```diff
@@ -6,3 +6,12 @@ tests:
...
       fixture: pwm_loopback
+  drivers.pwm.esp32.loopback:
+    tags: pwm drivers userspace
+    depends_on: pwm
+    filter: dt_compat_enabled("test-pwm-loopback")
+    harness: ztest
+    harness_config:
+      fixture: pwm_loopback
+    platform_allow: esp32
+    extra_args: "CONFIG_MCPWM_ESP32=y"
```
Comment (gmarull) [inline tests/drivers/pwm/pwm_loopback/testcase.yaml:17]: not required if enabling based on dt compat, see previous comment

**Review by gmarull — CHANGES_REQUESTED** [review]
a few nitpicks, otherwise lgtm

File: dts/bindings/pwm/espressif,esp32-mcpwm.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  Channel 6 -> Capture 0
+  Channel 7 -> Capture 1
+  Channel 8 -> Capture 2
+
+  Example: Use PWM0A output and capture 0:
+
+  pwm_loopback_0 {
+    compatible = "test-pwm-loopback";
+    pwms = <&mcpwm0 0 0 PWM_POLARITY_NORMAL>, #Channel 0 -> Output PWM0A
+           <&mcpwm0 6 0 PWM_POLARITY_NORMAL>; #Channel 6 -> Capture 0;
```
[the remark this answers is not in the document]
Comment (LucasTambor) [reply dts/bindings/pwm/espressif,esp32-mcpwm.yaml:39]: Yes. This example is taken from the pwm_loopback test, where the period property is 0 for both pwm and capture channels.

File: drivers/pwm/Kconfig.esp32 (hunk 2 above)
Comment (LucasTambor) [reply drivers/pwm/Kconfig.esp32:12]: OK

File: tests/drivers/pwm/pwm_loopback/testcase.yaml (hunk 3 above)
Comment (LucasTambor) [reply tests/drivers/pwm/pwm_loopback/testcase.yaml:17]: OK

**Review by sylvioalves — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**